### PR TITLE
notification schema

### PR DESCRIPTION
**Pull request by sina-jamshidi**

### CODE REVIEW

File: server/routes_src/index.ts [hunk 1]
```diff
@@ -12,4 +12,19 @@ router.post("/register", async (req, res, next) => {
   res.status(200).send({success: true, password: password});
 });
 
+router.get("/notification", async (req, res, next) => {
+  const notifications = await db.getNotifications(req.body.username);
+  res.status(200).send({success: true, notifications});
+});
```
Comment (frank1iu) [inline server/routes_src/index.ts]: Needs authentication
```suggestion
router.get("/notification", async (req, res, next) => {
  try {
    const user = await db.login(req.body.username, req.body.password);
    const notifications = await db.getNotifications(req.body.username);
    res.status(200).send({success: true, notifications});
  } catch (_) {
    res.status(304).send({success: false});
});
```

File: server/routes_src/index.ts [hunk 2]
```diff
@@ -12,4 +12,19 @@ router.post("/register", async (req, res, next) => {
...
+
+router.post("/notification", async (req, res, next) => {
+  const success = await db.addNotification(req.body.username, req.body.content, req.body.type as any);
+  res.status(200).send({success});
+});
```
Comment (frank1iu) [inline server/routes_src/index.ts:27]: I think server should generate notifications when there is a match (instead of client)

**Review by frank1iu — CHANGES_REQUESTED** [review]

File: server/routes_src/index.ts (hunk 1 above)
Comment (sina-jamshidi) [reply server/routes_src/index.ts]: oh we're actually doing that lol

File: server/routes_src/index.ts (hunk 2 above)
Comment (sina-jamshidi) [reply server/routes_src/index.ts:27]: Yup that makes sense. Just using it as a route for now for testing

File: server/routes_src/index.ts [hunk 3]
```diff
@@ -12,4 +12,23 @@ router.post("/register", async (req, res, next) => {
...
 
-export = router;
\ No newline at end of file
+router.get("/notification", async (req, res, next) => {
+  try {
+    const user = await db.login(req.body.username, req.body.password);
+    const notifications = await db.getNotifications(req.body.username);
+    res.status(200).send({success: true, notifications});
+  } catch (_) {
+    res.status(304).send({success: false});
```
Comment (frank1iu) [inline server/routes_src/index.ts:21]: not sure if it will work as i wrote it on github site

**Review by frank1iu — APPROVED** [review]

**Pull request merged**